pkg/runcfanotify/runcfanotify.go

```diff
@@ -199,14 +202,17 @@ func (n *RuncNotifier) watchContainerTermination(containerID string, containerPI
 	defer unix.Close(pidfd)
 
 	for {
+		if n.closed {
+			return
+		}
 		fds := []unix.PollFd{
 			{
 				Fd:      int32(pidfd),
 				Events:  unix.POLLIN,
 				Revents: 0,
 			},
 		}
-		count, err := unix.Poll(fds, -1)
+		count, err := unix.Poll(fds, 1000)
```

Comment:
Is there a way to poll on the fd and on a channel to avoid the timeout?

Reply:
If a node has 100 containers, that's 100 Go-routines that restart a Poll every second. Probably ok performance wise, but somewhat ugly.

Reply:
Ok, let's hold this PR for a bit then. I have some ideas to explore:
- Use a single goroutine to poll multiple FDs.
- Save the pidfds on a map and close them on `func (n *RuncNotifier) Close()` making Unix.Poll() return.

Reply:
If you poll multiple FDs, you will need to store the list in `struct RuncNotifier`. Then, this goes a bit in the direction of https://github.com/kinvolk/inspektor-gadget/pull/487#discussion_r871622918: you would need to keep track of containers. So that would simplify the commit 6c17fbfd7f33 about bundleDir.